core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -440,18 +440,31 @@ public RunContext forWorkerDirectory(ApplicationContext applicationContext, Work
         Map<String, Object> clone = new HashMap<>(this.variables);
 
         clone.put("workerTaskrun", clone.get("taskrun"));
-        if (clone.containsKey("parents")) {
-            clone.put("parents", clone.get("parents"));
-        }
-        if (clone.containsKey("parent")) {
-            clone.put("parent", clone.get("parent"));
-        }
 
         this.variables = ImmutableMap.copyOf(clone);
 
         return this;
     }
 
+    public RunContext forWorkerHandleDirectoryTask(ApplicationContext applicationContext, WorkerTask workerTask) {
```

Comment:
Why did you create a new method? Why don't you update the existing one as it is only used in worker so safe to update